src/api.js

```diff
@@ -69,7 +69,9 @@ router.patch('/users/:userID', (req, res, next) => {
 router.post('/users/:userID/checkIn', (req, res, next) => {
   req.db.UserModel.findOne({ id: req.params.userID })
     .then(user => {
-      user.checkIns.push({ pubName: req.body.pubName });
+      if (user.checkIns.filter(ci => ci.pubName === req.body.pubName).length === 0) {
```

Comment:
This is not idiomatic JavaScript - I'd much prefer `if(!(user.checkIns.some(ci => ci.pubName === req.body.pubName)))`

Reply:
Yeah we can do that.